### PR TITLE
Vault 2.x additional approle endpoints

### PR DESCRIPTION
**Pull request by beethan**
This is a follow up to #95 

With this #75 might be complete.

This implements a few commonly used AppRole endpoints including
- `upsert-role!` https://www.vaultproject.io/api-docs/auth/approle#create-update-approle
- `list-roles` https://www.vaultproject.io/api-docs/auth/approle#list-roles
- `read-role` https://www.vaultproject.io/api-docs/auth/approle#read-approle
- `read-role-id` https://www.vaultproject.io/api-docs/auth/approle#read-approle-role-id
- `generate-secret-id!` https://www.vaultproject.io/api-docs/auth/approle#generate-new-secret-id

I added a few additional tests, we may want to add a few more tests for these endpoints.

### CODE REVIEW

File: src/vault/auth/approle.clj [hunk 1]
```diff
@@ -24,6 +24,84 @@
     mount instead of the default. Passing `nil` will reset the client to the
     default.")
 
+  (upsert-role!
+    [client role-name opts]
+    "Creates a new AppRole or updates an existing AppRole.
+
+    When creating or updating an AppRole, there must be at least
+    one option supplied
```
Comment (greglook) [inline src/vault/auth/approle.clj:32]: Minor: several of the docstrings in this ns are missing periods at the end of sentences.

File: src/vault/auth/approle.clj [hunk 2]
```diff
@@ -24,6 +24,84 @@
     mount instead of the default. Passing `nil` will reset the client to the
     default.")
 
+  (upsert-role!
```
Comment (greglook) [inline src/vault/auth/approle.clj:27]: Naming: I think a different verb here would be clearer as to the purpose - how about `configure-role!` instead?

**Review by greglook — COMMENTED** [review]
Looks good to me - these will be useful for more programmatic interaction with app-roles. Two minor notes before merging.

**Comment by beethan** [comment]
This should be ready for 👀, sorry for the delay in getting to the feedback.

**Review by greglook — APPROVED** [review]
🙆‍♂️ :shipit:

**Pull request merged**